salt/states/junos.py

```diff
@@ -538,3 +528,49 @@ def commit_check(name):
     ret = {"name": name, "changes": {}, "result": True, "comment": ""}
     ret["changes"] = __salt__["junos.commit_check"]()
     return ret
+
+
+@resultdecorator
+def get_table(name, table, table_file, **kwargs):
```

Comment:
Needs a versionadded

   .. versionadded:: Sodium

Reply:
> Also need to now use pre-commit and black for PR submissions. They will be run as part of the PR test etc. but good to catch issues beforehand. Check https://github.com/saltstack/salt-enhancement-proposals/blob/master/accepted/0015-black.md where it was discussed.
> 
> Apart from changing published interfaces it looks good, published interfaces have to go through a deprecation cycle (used to be two major releases) if the changes are not backwards compatible, that is, can extend them, but should not affect existing parameters.

Used pre-commit tools and also done suggested changes like a single quote to double.
Also added dependencies list in Linux and darwin files.